### DISCUSSION TITLE
max upload size with superuser override

### DISCUSSION THREAD

**Discussion by eswanzey** [vanjarosoftware/Vanjaro.Platform#2087] [Ideas]
Vanjaro Platform defaults to 28MB max file uploads. In the security settings, the max value accepted is 28MB.

I increased the value of maxRequestLength and requestLengthDiskThreshold in the web.config from 28672 to 57344, to no apparent effect. In other words, I'm not seeing a way to increase the accepted size.

I like that the default is 28MB and that it can't be overridden by a site admin. In this case, I'm logged in as superuser but couldn't find a host-level setting if there is one. 

Even so, I would be more comfortable as a superuser/host if the max 28MB value was strictly enforced just like I experienced, BUT, provide the superuser login an "override" option that they can user when uploading, say, from the asset manager or the extension installer. In this way, the 1% of the time that you need to get a large file up there via the browser can be handled without opening the barn door wide enough to let in an elephant.

**Comment by hshimanshusharma** [comment]
Add the following parameters to the web.config file. For example, we are increasing the max upload size to 90MB. The value is in bytes:

```
 <system.webServer>
  <security>
      <requestFiltering>
        <requestLimits maxAllowedContentLength="93400320"/>
      </requestFiltering>
    </security>
  </system.webServer>
```

Also in the web. config, you will need to confirm that the maxRequestLength has been increased. The value is in kilobytes

`<httpRuntime targetFramework="4.7.2" shutdownTimeout="120" executionTimeout="1200" useFullyQualifiedRedirectUrl="true" maxRequestLength="91680" requestLengthDiskThreshold="91680" maxUrlLength="2048" requestPathInvalidCharacters="&lt;,&gt;,*,%,:,\,?" enableVersionHeader="false" requestValidationMode="2.0" fcnMode="Single" />`
